src/ts/components/settings/SettingsPanel.ts

```diff
@@ -66,6 +85,11 @@ export class SettingsPanel<Config extends SettingsPanelConfig> extends Container
   // navigation handling
   private activePage: SettingsPanelPage;
   private navigationStack: SettingsPanelPage[] = [];
+  
+  currentState: SettingsPanelState = null;
```

Comment:
Please explicitly state the visibility, in this case it seems to be only used internally so it can be `private`


```suggestion
  private currentState: SettingsPanelState = null;
```

Reply:
Ah good catch!